### PR TITLE
Fixes RHICOMPL-86: SystemPolicyCard ref_id truncates

### PR DESCRIPTION
**Pull request by akofink**
image

image

https://projects.engineering.redhat.com/browse/RHICOMPL-86

### CODE REVIEW

File: src/SmartComponents/Inventory/applications/compliance/SystemPolicyCard.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import React from 'react';
 import { CheckCircleIcon, ExclamationCircleIcon } from '@patternfly/react-icons';
+import Truncate from 'react-truncate';
```
Comment (karelhala) [inline src/SmartComponents/Inventory/applications/compliance/SystemPolicyCard.js]: Isn't this essentially the same as our long text truncate? [/src/PresentationalComponents/LongTextTooltip/LongTextTooltip.js](https://github.com/RedHatInsights/insights-frontend-components/blob/master/src/PresentationalComponents/LongTextTooltip/LongTextTooltip.js)

File: src/SmartComponents/Inventory/applications/compliance/SystemPolicyCard.js [hunk 2]
```diff
@@ -11,10 +12,16 @@ import {
 import PropTypes from 'prop-types';
 
 class SystemPolicyCard extends React.Component {
-    complianceIcon = () => {
-        const { policy: { compliant }} = this.props;
+    constructor(policy) {
+        super();
+        this.policy = policy.policy;
```
Comment (karelhala) [inline src/SmartComponents/Inventory/applications/compliance/SystemPolicyCard.js]: Please do not assign to class attributes, but use state. With this approach our component will not be re-rendered once policy is changed.

File: src/SmartComponents/Inventory/applications/compliance/SystemPolicyCard.js (hunk 1 above)
Comment (akofink) [reply src/SmartComponents/Inventory/applications/compliance/SystemPolicyCard.js]: This isn't a tooltip. Is that component misnamed or something?

**Comment by akofink** [comment]
Updated to only use `this.state`

File: src/SmartComponents/Inventory/applications/compliance/SystemPolicyCard.js (hunk 1 above)
Comment (ryelo) [reply src/SmartComponents/Inventory/applications/compliance/SystemPolicyCard.js]: I think @karelhala meant to link: https://github.com/RedHatInsights/insights-frontend-components/blob/master/doc/components/truncate.md

Comment (akofink) [reply src/SmartComponents/Inventory/applications/compliance/SystemPolicyCard.js]: Is there a way to use auto length? How am I supposed to know how many characters? That depends on the width of the browser, right?

**Comment by akofink** [comment]
What if it wraps instead of overflows? Is that okay?

File: src/SmartComponents/Inventory/applications/compliance/SystemPolicyCard.js [hunk 3]
```diff
@@ -11,10 +12,15 @@ import {
 import PropTypes from 'prop-types';
 
 class SystemPolicyCard extends React.Component {
-    complianceIcon = () => {
-        const { policy: { compliant }} = this.props;
+    constructor(policy) {
+        super();
```
Comment (karelhala) [inline src/SmartComponents/Inventory/applications/compliance/SystemPolicyCard.js]: You should pass the policy to parent as well. `super(policy)`.

File: src/SmartComponents/Inventory/applications/compliance/SystemPolicyCard.js [hunk 4]
```diff
@@ -11,10 +12,15 @@ import {
...
+        this.state = { refIdTruncated: <Truncate lines={ 1 }>{ policy.policy.ref_id }</Truncate>, ...policy };
+        this.onMouseover = this.onMouseover.bind(this);
```
Comment (karelhala) [inline src/SmartComponents/Inventory/applications/compliance/SystemPolicyCard.js]: No need to use binds.

File: src/SmartComponents/Inventory/applications/compliance/SystemPolicyCard.js [hunk 5]
```diff
@@ -23,29 +29,39 @@ class SystemPolicyCard extends React.Component {
             </div>;
     }
 
+    onMouseover() {
```
Comment (karelhala) [inline src/SmartComponents/Inventory/applications/compliance/SystemPolicyCard.js]: You should use arrow function to automatically bind `this`
```JS
onMouseOver = () => {...}
```

File: src/SmartComponents/Inventory/applications/compliance/SystemPolicyCard.js [hunk 6]
```diff
@@ -23,29 +29,39 @@ class SystemPolicyCard extends React.Component {
...
+        this.setState({ refIdTruncated: this.state.policy.ref_id });
+    }
+
+    onMouseout() {
```
Comment (karelhala) [inline src/SmartComponents/Inventory/applications/compliance/SystemPolicyCard.js]: You should use arrow function to automatically bind `this`
```JS
onMouseout= () => {...}
```

**Comment by akofink** [comment]
Updated to remove binds and pass policy to `super()`. I also refactored a card in our frontend in a similar way: https://github.com/RedHatInsights/compliance-frontend/pull/113

**Comment by karelhala** [comment]
Except for the truncate thing and @ryelo's request looks good! Thank you.

File: src/SmartComponents/Inventory/applications/compliance/SystemPolicyCard.js (hunk 1 above)
Comment (ryelo) [reply src/SmartComponents/Inventory/applications/compliance/SystemPolicyCard.js]: That's true. We may want to go back and revisit that issue with the other component. This should be fine for now.

**Comment by akofink** [comment]
Updated to wrap the untruncated text:

image

**Pull request merged**